config/domains.go

```diff
@@ -63,6 +74,16 @@ func (d Domain) GetCustomParams() url.Values {
 	return url.Values{}
 }
 
+// ToASCII converts the name of [Domain] to its ASCII form,
+// using non-transitional process specified in UTS 46.
+//
+// Note: conversion errors are silently discarded and partial conversion
+// results are used.
+func (d Domain) ToASCII() *Domain {
+	d.DomainName, _ = nontransitionalLookup.ToASCII(d.DomainName)
```

Comment:
The `d.DomainName` has been changed also.

Reply:
The receiver of this function is a value, not a pointer. So changing `d.DomainName` here will not have any side effects.

See: https://go.dev/doc/faq#methods_on_values_or_pointers

Reply:
I didn't pay attention.  but It can simply return the character instead of the object.

Reply:
Does `SubDomain` need to change, example`d.SubDomain, _ = nontransitionalLookup.ToASCII(d.SubDomain)`

Reply:
So it looks like we can now just hardcode it to `String()` depending on usage.

Reply:
okay, If there are any other requirements later, we will change it then.